Add documentation for Atag integration (#11841)
home-assistant/home-assistant.github.io: source/_integrations/atag.markdown

---
title: Atag
description: Instructions on how to setup Atag integration.
logo: atag.png
ha_category:
  - Climate
  - Water heater
  - Sensor
ha_release: 0.105
ha_iot_class: Local Polling
---

The `Atag` integration allows Home Assistant to connect to [Atag One](https://atag-one.com) thermostats, reporting and setting its status.
The integration implements the following platforms.

- Climate
- Water Heater
- Sensor

## Configuration

> The Atag integration can be enabled directly from the config flow UI.

Reviewer: For a user the "config flow UI" is not something that exists, those are internal names use in the development.

Please describe, from a user perspective, how to set this up.

In a regular setup only the IP address has to be provided (UDP discovery planned for future release).

> Alternatively the integration can be setup through configuration.yaml.

Reviewer: ```suggestion
Alternatively, the integration can be set up through the `configuration.yaml` file.
```

> This way additional (non-default) sensors can be enabled. Only the `host` field is required.

Reviewer: Comma:
```suggestion
This way, additional (non-default) sensors can be enabled. Only the `host` field is required.
```

Furthermore, this sounds weird. Why would a manual configuration via YAML give more sensors?

```yaml
# Minimal setup through configuration.yaml
atag:
  host: YOUR_ATAG_ADDRESS
```

{% configuration %}
host:

>   description: Atag host or ip address.

Reviewer: ```suggestion
  description: Atag host or IP address.
```

  required: true
  type: string
email:
  description: Email registered in the Atag App.
  required: false
  type: string
port:

>   description: Port to reach the Atag Api. Only needed if connecting through alternative routes.

Reviewer: ```suggestion
  description: Port to reach the Atag API. Only needed if connecting through alternative routes.
```

  required: false
  type: integer
scan_interval:
  description: Polling frequency in seconds. Defaults to 30, increasing may overload the device.
  required:  false
  type: integer
sensors:

>   description: List of sensors to add. Uses default list if not specified. See below for the full list.

Reviewer: We used to call this "MONITORED_CONDITIONS", which we try to remove/avoid nowadays. Make sure your integration adhere to: <https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md>

If you want to prevent all sensor showing up when initially setting up the integration, Home Assistant offers a way to disable non-popular sensors by default, allowing the user to enable them if they wish. See: <https://developers.home-assistant.io/docs/en/next/entity_registry_disabled_by.html#integrations-setting-default-value-of-disabled_by-for-new-entity-registry-entries>

Author: OK - this will require some changes in the way the component is currently set up. Will make the necessary changes in the coming days

  type: list
  required: false
{% endconfiguration %}

<div class='note warning'>

> Setting `scan-interval` below 30 may overload the device. Make sure to use the default if you experience difficulties (e.g. network disconnects)

Reviewer: Missing comma and full stop. Needs empty lines before and after to allow the Markdown parser to switch context between Markdown & HTML.
```suggestion

Setting `scan-interval` below 30 may overload the device. Make sure to use the default if you experience difficulties (e.g., network disconnects).

```

Just wondering; Why would we offer a `scan-interval`, to begin with?

Author: Agree, that level of configurability may not be necessary. I'll remove it as an option.

</div>

<div class='note'>

> Changes to configuration.yaml may not be picked up automatically. Make sure to manually delete the config entry in the UI in case your configuration is not updated.

Reviewer: ```suggestion
Changes to `zconfiguration.yaml` may not be picked up automatically. Make sure to manually delete the config entry in the UI in case your configuration is not updated.
```

> Changes to configuration.yaml may not be picked up automatically. Make sure to manually delete the config entry in the UI in case your configuration is not updated.

Reviewer: That is weird behavior, we should fix that on the integration level, not in the documentation.

Author: True, I was looking for a clean way to resolve this. Various approaches across integrations - is there any preferred solution / example regarding `configuration.yaml` changes and existing config entries?

</div>

## Climate

The `Atag` climate platform provides current and target temperature information for the heating system, boiler status and HVAC mode.

### Component services

This integration supports the following services (see [Climate](/integrations/climate/)).

- [`set_temperature`](/integrations/climate/#service-climateset_temperature)
- [`set_hvac_mode`](/integrations/climate/#service-climateset_hvac_mode)
  - `heat` for regular thermostat mode
  - `auto` for weather based mode
  - `off` to disable control from Home Assistant

<div class='note'>

> HVAC mode provides returns regular or weather-based mode. (i.e. not manual, auto, extend, fireplace).

Reviewer: ```suggestion
HVAC mode provides returns regular or weather-based mode (i.e., not manual, auto, extend, fireplace).
```

The hold modes (manual, auto, extend) are currently available as a sensor only. Setting these modes is planned for a future update.
</div>

## Water Heater

> The Water Heater reports current and target temperature for Domestic Hot Water demand, as well as boiler status (heating or idle). This can be used to detect hot water demand, such a running shower or tap water.

Reviewer: ```suggestion
The Water Heater reports current and target temperature for Domestic Hot Water demand, as well as boiler status (heating or idle). This can be used to detect hot water demand, such as a running shower or tap water.
```

Setting target values is currently not supported.

## Sensor

> The `Atag` sensor platform provides additional metrics reported by the One, which are not part of either the Water-Heater or Climate platforms.

Reviewer: ```suggestion
The `Atag` sensor platform provides additional metrics reported by the One, which is not part of either the Water-Heater or Climate platforms.
```

Author: This I'll change as per the guidelines you posted earlier.

Author: Rewrote this because `are` actually refers to the metrics - which is plural. Should be more clear now.

> By default the following sensors are enabled.

Reviewer: ```suggestion
By default, the following sensors are enabled:
```

```yaml
sensors:
  - outside_temp
  - outside_temp_avg
  - weather_status
  - operation_mode
  - ch_water_pressure
  - dhw_water_temp
  - dhw_water_pres
  - burning_hours
  - flame_level
```

> Alternatively the user can provide a list of sensors to enable (requires setup through `configuration.yaml`).

Reviewer: ```suggestion
Alternatively, the user can provide a list of sensors to enable (requires setup through `configuration.yaml`).
```

logo: atag.png
ha_category:
  - Climate
  - Water heater
  - Sensor
ha_release: 0.105
ha_iot_class: Local Polling
---

The `Atag` integration allows Home Assistant to connect to [Atag One](https://atag-one.com) thermostats, reporting and setting its status.
The integration implements the following platforms:

- Climate
- Water Heater
- Sensor

## Configuration

The Atag integration can be enabled directly from Home Assistant. Navigate to `configuration`, then `integrations` and click `add`. Click `Atag` to initiate the configuration. Only the IP Address has to be provided, but be sure to add your email address if you experience connectivity issues.

{% configuration %}
host:
  description: Atag hostname or IP address.
  required: true
  type: string
email:
  description: Email registered in the Atag App.
  required: false
  type: string
port:
  description: Port to reach the Atag API. Only needed if connecting through alternative routes.
  required: false
  type: integer
{% endconfiguration %}

## Climate

The `Atag` climate platform provides current and target temperature information for the heating system, boiler status and HVAC mode.

> ### Component services

Reviewer: We don't use the term "component" when user-facing content. Use "Integration" instead.
```suggestion
### Integration services
```

This integration supports the following services (see [Climate](/integrations/climate/)).

- [`set_temperature`](/integrations/climate/#service-climateset_temperature)
- [`set_hvac_mode`](/integrations/climate/#service-climateset_hvac_mode)
  - `heat` for regular thermostat mode

>   - `auto` for weather based mode

Reviewer: ```suggestion
  - `auto` for weather-based mode
```
